src/test/java/com/emotion/emotiondesktop/Integration/HttpRequestTest.java

```diff
@@ -0,0 +1,68 @@
+package com.emotion.emotiondesktop.Integration;
+
+import com.emotion.emotiondesktop.Helper.HttpRequest;
+import com.emotion.emotiondesktop.Helper.HttpResponse;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import java.io.ByteArrayInputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+class HttpRequestTest {
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @Test
+    void testSendHttpRequest_Success() throws Exception {
+        String apiUrl = "https://dev.escooters.blumilk.pl/api/login";
+        String method = "POST";
+        String email = "admin@example.com";
+        String password = "password";
+        String jsonInputString = "{\"email\": \"" + email + "\", \"password\": \"" + password + "\"}";
+        int responseCode = HttpURLConnection.HTTP_OK;
+
+        URL mockUrl = mock(URL.class);
+        HttpURLConnection mockConnection = mock(HttpURLConnection.class);
+
+        when(mockUrl.openConnection()).thenReturn(mockConnection);
+
+        when(mockConnection.getResponseCode()).thenReturn(responseCode);
+
+        HttpResponse response = HttpRequest.sendHttpRequest(apiUrl, method, jsonInputString);
+
+        assertEquals(responseCode, response.getResponseCode());
+    }
+
+    @Test
+    void testSendHttpRequest_Failure() throws Exception {
```

Comment:
I see that you changed tests naming. Is some convention in Java?

Reply:
For me some tests name are not clear, like `testUserLogIn_WhenInvalidEmailFormat` - it took a while to think that user cannot login because of invalid email format

Reply:
changed